[CONTRIBUTING.md] A note about the use of AI tools (#15010)
Qiskit/qiskit: CONTRIBUTING.md

sure to always be kind and respectful in your interactions with maintainers and other contributors, you can read
[the Qiskit Code of Conduct](https://github.com/Qiskit/qiskit/blob/main/CODE_OF_CONDUCT.md).

### Use of AI-assisted tools

> If you use AI-assisted tools when preparing code, remember that your contribution must still be your own original work of authorship, as required by the [Contributor License Agreement (CLA)](https://qisk.it/cla).

Reviewer: This is a bit nitpicky, but I don't think the term "AI-assisted tools" is very widely used, so I would say simply "AI tools":

```suggestion
### Use of AI tools

If you use AI tools while preparing your contribution, please ensure that your code remains your own original work of authorship, as required by the [Contributor License Agreement (CLA)](https://qisk.it/cla).
```

> Make sure you fully understand, review, and take responsibility for the code you submit, and be able to explain the reasoning behind it during review.

Reviewer: ```suggestion
Make sure you fully understand, review, and take responsibility for the code you submit. You must be able to explain the reasoning behind it during review.
```

Submissions that appear unreviewed or copied directly from an AI assistant without proper understanding may be requested to be revised or declined.

> Please also note that spamming issues or pull requests with AI-generated comments is strictly prohibited under the (Qiskit Code of Conduct](https://qisk.it/coc).

Author: ```suggestion
Please also note that spamming issues or pull requests with AI-generated comments is strictly prohibited under the [Qiskit Code of Conduct](https://qisk.it/coc).
```

sure to always be kind and respectful in your interactions with maintainers and other contributors, you can read
[the Qiskit Code of Conduct](https://github.com/Qiskit/qiskit/blob/main/CODE_OF_CONDUCT.md).

### Use of AI tools

> [!WARNING]
> If you use any AI tool while preparing your code contribution, you **must** disclose the name of the tool and its version in the PR description.

When using AI tools for code generation, your submission must still be your own original work of authorship, as required by the [Contributor License Agreement (CLA)](https://qisk.it/cla). It is your responsibility to make sure that:

- You review and fully understand the generated code, and you can explain the reasoning behind it during review.
- The usage of the AI tool does not violate any third-party license obligations.

> - The AI tool’s terms and conditions allow its output to be used in open source projects and are compatible with the [Qiskit license](LICENSE.txt), the [Qiskit CLA](https://qisk.it/cla), and [these Contributor Guidelines](CONTRIBUTING.md).

Reviewer 2: ```suggestion
- The AI tool's terms and conditions allow its output to be used in open source projects and are compatible with the [Qiskit license](LICENSE.txt), the [Qiskit CLA](https://qisk.it/cla), and [these Contributor Guidelines](CONTRIBUTING.md).
```

kind of weird to have smart quotes in a basic text file

Author: hmm.. maybe wrong keyboard layout? good catch

- You only use AI tools that have features to:
  * filter out generated code substantially similar to training data, or

>   * identify similar training code so you can comply with the original license obligations (notice, attribution, etc.) and only contribute if it’s compatible with the [Qiskit license](LICENSE.txt).

Reviewer 2: ```suggestion
  * identify similar training code so you can comply with the original license obligations (notice, attribution, etc.) and only contribute if it's compatible with the [Qiskit license](LICENSE.txt).
```

same

>   * identify similar training code so you can comply with the original license obligations (notice, attribution, etc.) and only contribute if it’s compatible with the [Qiskit license](LICENSE.txt).

Reviewer 2: Not saying I necessarily have a problem with these conditions, but I don't remember the discussion about them, and I don't know which AI tooling has this and which doesn't - do you have any examples?

If we're really not sure on this at all, it's probably best not to include it - I think the most important thing is trying to collate LLM disclosures, so we can trace them in the future if necessary.
